client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,300 @@
+        pipeline = RegisteredPipeline(
+            graph=make_mock_pipeline_graph(),
+            registered_model_version=rmv,
+        )
+    pipeline._log_pipeline_definition_artifact()
+
+
+def test_get_pipeline_definition_artifact(
+    make_mock_registered_model_version,
+    make_mock_simple_pipeline_definition,
+) -> None:
+    """Test that a pipeline definition artifact can be fetched from the
+    registered model version associated with a RegisteredPipeline object.
+    """
+    rmv = make_mock_registered_model_version()
+    pipeline_definition = RegisteredPipeline._get_pipeline_definition_artifact(
+        registered_model_version=rmv,
+    )
+    assert pipeline_definition == make_mock_simple_pipeline_definition(id=rmv.id)
+
+
+def test_to_pipeline_definition(
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that a pipeline definition can be constructed from a
+    RegisteredPipeline object.
+
+    In depth testing of the `_to_graph_definition`
+    and `to_steps_definition` functions are handled in unit tests for
+    PipelineGraph.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+    pipeline_definition = pipeline._to_pipeline_definition()
+    assert pipeline_definition == {
+        "pipeline_version_id": pipeline.id,
+        "graph": graph._to_graph_definition(),
+        "predecessors": graph._to_steps_definition(),
+    }
+
+
+@given(resources=resources())
+def test_to_pipeline_configuration_valid(
+    resources,
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that a pipeline configuration can be constructed from a
+    RegisteredPipeline object and a valid list of pipeline resources.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        step_resources = {step.name: resources for step in graph.steps}
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+
+    pipeline_configuration = pipeline._to_pipeline_configuration(
+        pipeline_resources=step_resources
+    )
+    assert pipeline_configuration["pipeline_version_id"] == pipeline.id
+    for graph_step, config_step in zip(graph.steps, pipeline_configuration["steps"]):
+        # All steps are included in the configuration
+        assert graph_step.name == config_step["name"]
```

Comment:
This doesn't actually check _all_ graph steps are in the config, since `zip()` truncates the shorter list.

Reply:
Good catch.  I threw an equal length assertion into this test and added a new test for scenarios when resources are provided for some, but not all, of the steps in the pipeline.